### ISSUE TITLE
API_IMAGE_LAYOUT error

### ISSUE THREAD

**Issue by SaTjemkes** [GenericMappingTools/gmt-python#232]
Hi

After a new build using the latest material of gmt and gmt-python from the repository the gmt module was so kind to throw me out after providing the message listed below.

If you could guide me how to resolve this it would be appreciated.

stephen

**Description of the problem**

**Full code that generated the error**

```python
(hidpf) python3
Python 3.6.3 (default, Aug 30 2018, 17:20:23) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import gmt
gmt-python-session [ERROR]: Syntax error: Unrecognized keyword API_IMAGE_LAYOUT
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/stephen/gmt/gmt-python/gmt/__init__.py", line 25, in <module>
    _begin()
  File "/home/stephen/gmt/gmt-python/gmt/session_management.py", line 17, in begin
    with Session() as lib:
  File "/home/stephen/gmt/gmt-python/gmt/clib/session.py", line 175, in __enter__
    version = self.info["version"]
  File "/home/stephen/gmt/gmt-python/gmt/clib/session.py", line 154, in info
    "image layout": self.get_default("API_IMAGE_LAYOUT"),
  File "/home/stephen/gmt/gmt-python/gmt/clib/session.py", line 447, in get_default
    name, status
gmt.exceptions.GMTCLibError: Error getting default value for 'API_IMAGE_LAYOUT' (error code 67).
```

**Full error message**

```
PASTE ERROR MESSAGE HERE
```

**System information**

* Operating system:
* Python installation (Anaconda, system, ETS):
* Version of GMT:
* Version of Python:
* Version of this package:
* If using conda, paste the output of `conda list` below:

<details>
<summary>output of conda list</summary>
<pre>
PASTE OUTPUT OF CONDA LIST HERE
</pre>
</details>

**Comment by leouieda** [comment]
@SaTjemkes please use the template provided when reporting an issue. The fields there are all necessary for tracking down problems. I copied the template back into your answer. Please use the Edit button (click on the `...` on the top right of the issue description) to fill out all the fields.

**Comment by SaTjemkes** [comment]
Of course,
Was not aware of a template, could this be automatically copied once you activate the report an issue?

Thanks
stephjenm

**Comment by SaTjemkes** [comment]
Full error message
gmt.exceptions.GMTCLibError: Error getting default value for 'API_IMAGE_LAYOUT' (error code 67).



System information
•Operating system: Lubuntu 18.0-4
•Python installation (Anaconda, system, ETS):  complied from source,
•Version of GMT: The latest on 30 August 2018 (started from a git clone etc)
•Version of Python: 3.6.3, compiled using gcc 7.3.0. 
•Version of this package: the latest on 30 August 2018 started from a git clone
•If using conda, paste the output of conda list below:

**Comment by leouieda** [comment]
@SaTjemkes sorry for the long delay. That's a strange error and I've never had anyone who tried running on a self-compiled Python so that's great for diversity! Could you post the output of `gmt --version` from the command-line? GMT is not great about having every commit tested so sometimes they commit things that are broken.

**Comment by leouieda** [comment]
@SaTjemkes thanks again for the feedback! This has been fixed in #235. Would you mind updating GMT/Python to see if the error persists?

**Comment by SaTjemkes** [comment]
Dear Leonardo,

No problem however, as I want to test on the most recent version of everything,  there is a persistent issue.

I can find the gmt-python distro, and cloned it, but it requires gmt-6.x.x, and this is the difficult bit.

My suggestion put a few lines in your requirements.txt (or in your README.rst at the bottom) , such that naïve users like me don’t have to spend hours re-inventing the wheel.

I did clone: git clone https://github.com/GenericMappingTools/gmt.git

But it is unclear to me if this is version 6 or not (I assume not) there is nothing in the README.rst about version

And on the ftp server there is the 5.4.4 version

stephen

**Comment by SaTjemkes** [comment]
Hi Leonardo

I have to apologize, installation instructions are on the web page of the gmt-python. I simply read over it.

The good news is that indeed the system seems to be working nicely.
What I did

I installed gdal from source,
Cloned the gmt distubution from github, compiled and installed it on my environment (including the 2 extra distributions for coastlines etc)
(in a dedicted directory myhome/.local/gmt)
I cloned the latest gmt-python code from github, build, and installed it on my environment
Added the myhome/.local/gmt/bin to search path, myhome/.local/gmt/lib to LD_LIBRSRY_PATH
Defined a GMT_LIBRARY_PATH =/myhome/.local/gmt/lib
(exported all environmental variables)

started a jupyter notebook
and ran one of my simple examples by which I plot a circle on a Mercator map showing the location of gruan radiosonde stations

This showed up nicely, proof is enclosed

Thank you for the effort

And apologise for confusion

stephen
